agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wascore/statsbeat/AzureMetadataService.java

```diff
@@ -94,24 +93,43 @@ void parseJsonResponse(String response) throws IOException {
   public void run() {
     HttpRequest request = new HttpRequest(HttpMethod.GET, ENDPOINT);
     request.setHeader("Metadata", "true");
+    HttpResponse response;
     try {
-      HttpResponse response = LazyHttpClient.getInstance().send(request).block();
-      if (response == null) {
-        // this shouldn't happen, the mono should complete with a response or a failure
-        throw new AssertionError("http response mono returned empty");
+      response = LazyHttpClient.getInstance().send(request).block();
+    } catch (Exception ex) {
+      if (ex.getCause()
+          .toString()
+          .contains("Network is unreachable: no further information: /169.254.169.254:80")) {
```

Comment:
Is there any other way we can identify that the exception is due to not running on vm or vmss?

Reply:
socket exception. this end point doesn't go out to the internet because it's a local linked ip.. non routable.. we are good to assume any exceptions here are likely caused by the azure metadata service.